simpleclient_dropwizard/src/main/java/io/prometheus/client/dropwizard/samplebuilder/MapperConfig.java

```diff
@@ -0,0 +1,154 @@
+ * Dropwizard metrics that match the "match" pattern will be further processed to have a new name and new labels based on this config.
+ */
+public class MapperConfig {
+    // each part of the metric name between dots
+    private static final String METRIC_PART_REGEX = "[a-zA-Z_0-9](-?[a-zA-Z0-9_])+";
+    // Simplified GLOB: we can have "*." at the beginning and "*" only at the end
+    static final String METRIC_GLOB_REGEX = "^(\\*\\.|" + METRIC_PART_REGEX + "\\.)+(\\*|" + METRIC_PART_REGEX + ")$";
+    // Labels validation.
+    private static final String LABEL_REGEX = "^[a-zA-Z_][a-zA-Z0-9_]+$";
+    private static final Pattern MATCH_EXPRESSION_PATTERN = Pattern.compile(METRIC_GLOB_REGEX);
+    private static final Pattern LABEL_PATTERN = Pattern.compile(LABEL_REGEX);
+
+    /**
+     * Regex used to match incoming metric name.
+     * Uses a simplified glob syntax where only '*' are allowed.
+     * E.g:
+     * org.company.controller.*.status.*
+     * Will be used to match
+     * org.company.controller.controller1.status.200
+     * and
+     * org.company.controller.controller2.status.400
+     */
+    private String match;
+
+    /**
+     * New metric name. Can contain placeholders to be replaced with actual values from the incoming metric name.
+     * Placeholders are in the ${n} format where n is the zero based index of the group to extract from the original metric name.
+     * E.g.:
+     * match: test.dispatcher.*.*.*
+     * name: dispatcher_events_total_${1}
+     * <p>
+     * A metric "test.dispatcher.old.test.yay" will be converted in a new metric with name "dispatcher_events_total_test"
+     */
+    private String name;
+
+    /**
+     * Labels to be extracted from the metric name.
+     * They should contain placeholders to be replaced with actual values from the incoming metric name.
+     * Placeholders are in the ${n} format where n is the zero based index of the group to extract from the original metric name.
+     * E.g.:
+     * match: test.dispatcher.*.*
+     * name: dispatcher_events_total_${0}
+     * labels:
+     * label1: ${1}_t
+     * <p>
+     * A metric "test.dispatcher.sp1.yay" will be converted in a new metric with name "dispatcher_events_total_sp1" with label {label1: yay_t}
+     * <p>
+     * Label names have to match the regex ^[a-zA-Z_][a-zA-Z0-9_]+$
+     */
+
+    private Map<String, String> labels = new HashMap<String, String>();
+
+    public MapperConfig() {
+        // empty constructor
+    }
+
+    // for tests
+    MapperConfig(final String match) {
+        validateMatch(match);
+        this.match = match;
+    }
+
+    public MapperConfig(final String match, final String name, final Map<String, String> labels) {
+        this.name = name;
+        validateMatch(match);
+        this.match = match;
+        validateLabels(labels);
+        this.labels = labels;
+    }
+
+    public String getMatch() {
+        return match;
+    }
+
+    public void setMatch(final String match) {
```

Comment:
Do you need all these given that it's coming from the constructor now?

Reply:
I left the setters so that it can be bound to SpringBoot configuration properties